src/Compilers/CSharp/Test/Syntax/Parsing/RecordParsing.cs

```diff
@@ -408,6 +409,657 @@ public void RecordParsingAmbiguities()
+                // (1,60): error CS1514: { expected
+                // abstract class C<T1, T2> where T1 : class where T2 : class;
+                Diagnostic(ErrorCode.ERR_LbraceExpected, "").WithLocation(1, 60),
+                // (1,60): error CS1513: } expected
+                // abstract class C<T1, T2> where T1 : class where T2 : class;
+                Diagnostic(ErrorCode.ERR_RbraceExpected, "").WithLocation(1, 60)
+                );
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.ClassDeclaration);
+                {
+                    N(SyntaxKind.AbstractKeyword);
+                    N(SyntaxKind.ClassKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.TypeParameterList);
+                    {
+                        N(SyntaxKind.LessThanToken);
+                        N(SyntaxKind.TypeParameter);
+                        {
+                            N(SyntaxKind.IdentifierToken, "T1");
+                        }
+                        N(SyntaxKind.CommaToken);
+                        N(SyntaxKind.TypeParameter);
+                        {
+                            N(SyntaxKind.IdentifierToken, "T2");
+                        }
+                        N(SyntaxKind.GreaterThanToken);
+                    }
+                    N(SyntaxKind.TypeParameterConstraintClause);
+                    {
+                        N(SyntaxKind.WhereKeyword);
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "T1");
+                        }
+                        N(SyntaxKind.ColonToken);
+                        N(SyntaxKind.ClassConstraint);
+                        {
+                            N(SyntaxKind.ClassKeyword);
+                        }
+                    }
+                    N(SyntaxKind.TypeParameterConstraintClause);
+                    {
+                        N(SyntaxKind.WhereKeyword);
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "T2");
+                        }
+                        N(SyntaxKind.ColonToken);
+                        N(SyntaxKind.ClassConstraint);
+                        {
+                            N(SyntaxKind.ClassKeyword);
+                        }
+                    }
+                    M(SyntaxKind.OpenBraceToken);
+                    M(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact, WorkItem(45538, "https://github.com/dotnet/roslyn/issues/45538")]
+        public void AbstractMethod_ConstrainsAndSemiColon()
+        {
+            UsingTree("abstract class C { abstract void M<T>() where T : class; }");
```

Comment:
>abstract class C { abstract void M<T>() where T : class; } [](start = 23, length = 58)

Is this a new scenario? We must have many tests that hit this already.

Reply:
Consider changing the test to `abstract record R { abstract void M<T>() where T : class; }`

---
In reply to: [456567211](https://github.com/dotnet/roslyn/pull/45943#discussion_r456567211) [](ancestors = 456567211)